### PR TITLE
ENYO-6227: Fix FormCheckboxItem styling of `itemIcon`

### PR DESCRIPTION
**Pull request by Djspaceg**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
Items change size when an icon is added

### Resolution
Remove the margin of child icons which bump the item to be taller than intended.
Also sets the block sizing so the new span element has the correct dimensions.

### CODE REVIEW

File: packages/moonstone/FormCheckboxItem/FormCheckboxItem.module.less [hunk 1]
```diff
@@ -24,6 +24,15 @@
 		overflow: hidden;
 	}
 
+	.itemIcon {
+		display: inline-block;
+
+		* {
```
Comment (ryanjduffy) [inline packages/moonstone/FormCheckboxItem/FormCheckboxItem.module.less]: Good catch! Can we be more conservative and only affect children with `> *` as is done in SlotItem?

Comment (Djspaceg) [reply packages/moonstone/FormCheckboxItem/FormCheckboxItem.module.less]: I was concerned about nested elements, say you have a DIV or a SPAN with icons in it. Those icons wouldn't get trimmed and you wouldn't know why they don't fit. DIV and SPAN may still be commonly used if developers are unaware of `React.createFragment`.

File: packages/moonstone/FormCheckboxItem/FormCheckboxItem.module.less [hunk 2]
```diff
@@ -24,6 +24,15 @@
 		overflow: hidden;
 	}
 
+	.itemIcon {
+		display: inline-block;
```
Comment (ryanjduffy) [inline packages/moonstone/FormCheckboxItem/FormCheckboxItem.module.less]: Might want to use `display: contents` here. Noticing that the left margins collapse in 3.0.0 but not any longer with the current code and this fix.

nebula:

image

this:

image

File: packages/moonstone/FormCheckboxItem/FormCheckboxItem.module.less (hunk 1 above)
Comment (ryanjduffy) [reply packages/moonstone/FormCheckboxItem/FormCheckboxItem.module.less]: Similarly, if devs put non-`moonstone/Icon`s there, every node would lose its margins which would be equally strange.

**Pull request merged**